tests/azureml/test_image_creation.py

```diff
@@ -390,10 +421,50 @@ def test_execution_script_init_method_attempts_to_load_correct_azure_ml_model(
         assert get_model_path_call_kwargs["version"] == model_version
 
 
-def test_execution_script_run_method_scores_pandas_dataframes_successfully(
-        sklearn_model, model_path):
+def test_execution_script_run_method_scores_pandas_dfs_successfully_when_model_outputs_numpy_arrays(
+        sklearn_model, sklearn_data, model_path):
     mlflow.sklearn.save_model(sk_model=sklearn_model, path=model_path)
 
+    pyfunc_model = mlflow.pyfunc.load_pyfunc(path=model_path)
+    pyfunc_outputs = pyfunc_model.predict(sklearn_data[0])
+    assert isinstance(pyfunc_outputs, np.ndarray)
+
+    model_mock = Mock()
+    model_mock.name = "model_name"
+    model_mock.version = 1
+
+    with TempDir() as tmp:
+        execution_script_path = tmp.path("dest")
+        mlflow.azureml._create_execution_script(
+                output_path=execution_script_path, azure_model=model_mock)
+
+        with open(execution_script_path, "r") as f:
+            execution_script = f.read()
+
+    # Define the `init` and `score` methods contained in the execution script
+    # pylint: disable=exec-used
+    exec(execution_script, globals())
+    with AzureMLMocks() as aml_mocks:
+        aml_mocks["get_model_path"].side_effect = lambda *args, **kwargs: model_path
+        # Execute the `init` method of the execution script and load the sklearn model from the
+        # mocked path
+        # pylint: disable=undefined-variable
+        init()
+
+        # Invoke the `run` method of the execution script with sample input data and verify that
+        # reasonable output data is produced
+        # pylint: disable=undefined-variable
+        output_data = run(pd.DataFrame(data=sklearn_data[0]).to_json(orient="split"))
+        assert len(output_data) == len(sklearn_data[0])
```

Comment:
I am not sure if comparing length is good enough. I would rather compare we get the same results as from calling pyfunc ourselves.